src/sql/migrations.rs

```diff
@@ -579,6 +579,19 @@ CREATE INDEX smtp_messageid ON imap(rfc724_mid);
         )
         .await?;
     }
+    if dbversion < 86 {
+        info!(context, "[migration] v86");
+        sql.execute_migration(
+            r#"CREATE TABLE bobstate (
+                   id INTEGER PRIMARY KEY AUTOINCREMENT,
```

Comment:
What about autoincrement here, is it needed? Was this added before we looked up the documentation or is it added on purpose?

Reply:
This was on purpose, in bobstate.rs `.update_next()` updates or deletes based on an ID it got from an earlier query.  This would not be possible safely without autoincrement.

> Was this added before we looked up the documentation

I'm confused by this, I recall not too long ago reviewing a PR from @r10s and noting that autoincrement was not needed to which the response was "we do this everywhere just to be sure".  So what is our current practice?

AFAIK the main downside is that this doubles (or more?) the number of transactions on a table using it.  For this table this shouldn't be much of a problem, maybe for some higher-use tables it matters more.

Reply:
Let's keep it here then since it's needed.

There is no best practice, for imap table it probably makes sense to avoid for performance and for msgs table it's better to have autoincrement to avoid bugs like sending a draft to the wrong chat accidentally because the UI decided to cache the msgid and it was reused.